src/models/highlightedtext.h

```diff
@@ -0,0 +1,67 @@
+/*
+    SPDX-FileCopyrightText: Lieven Hey <lieven.hey@kdab.com>
+    SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+
+    SPDX-License-Identifier: GPL-2.0-or-later
+*/
+
+#pragma once
+
+#include <QObject>
+#include <QTextLayout>
+#include <QTextLine>
+
+#include <KColorScheme>
+
+#include "hotspot-config.h"
+
+namespace KSyntaxHighlighting {
+class SyntaxHighlighter;
+class Definition;
+class Repository;
+}
+
+class HighlightingImplementation
+{
+    Q_DISABLE_COPY(HighlightingImplementation)
+public:
+    HighlightingImplementation() = default;
+    virtual ~HighlightingImplementation() = default;
```

Comment:
move the `= default` to the .cpp - we don't want to inline the vtable into every TU that includes this header